### ISSUE TITLE
Replace all calls to rand() by calls to a custom randomizer

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/virtualc64#788]
VirtualC64 calls `rand()` at multiple places. Although the returned sequences are deterministic, this causes issues in run-ahead mode, since both instances would query different numbers and diverge. 

TODO: Implement a new Randomizer class that returns a deterministic pseudo-random sequence and instantiate this class in both the main instance and the run-ahead instance.

Fun fact: Bugs related to run-ahead mode result in funny behavior due to diverging emulator instances. In Barry McGuigan World Championship Boxing, I've thought to have knocked out my opponent. 

<img width="918" alt="Bildschirmfoto 2024-03-05 um 17 38 20" src="https://github.com/dirkwhoffmann/virtualc64/assets/12561945/a0654c48-1fd7-40c6-92fc-6a6db759f608">

However, this only happened in the run-ahead instance. After the next resync, I've been lying on the ground and lost the fight.

**Comment by dirkwhoffmann** [comment]
An easy solution is to derive the pseudo-number directly from the current CPU cycle:

```C++
u32
C64::random()
{
    return random(u32(cpu.clock));
}

u32
C64::random(u32 seed)
{
    // Parameters for the Linear Congruential Generator (LCG)
    u64 a = 1664525;
    u64 c = 1013904223;
    u64 m = 1LL << 32;

    // Apply the LCG formula
    return u32((a * seed + c) % m);
}
```

`random()` can be used as long as there is only one call per cycle. Otherwise, the same value would be returned twice. 

Arrays such as `colorRam` can be initialized as follows:

```C++
        // Initialize color RAM with random numbers
        u32 seed = 0;
        for (isize i = 0; i < isizeof(colorRam); i++) {
            
            seed = c64.random(seed);
            colorRam[i] = u8(seed);
        }
```

**Comment by dirkwhoffmann** [comment]
Fixed in v5.0b1
